var/spack/repos/builtin/packages/esmf/package.py

```diff
@@ -249,12 +289,10 @@ def setup_build_environment(self, env):
             elif "^mvapich2" in spec:
                 env.set("ESMF_COMM", "mvapich2")
             elif "^mpich" in spec:
-                # esmf@7.0.1 does not include configs for mpich3,
-                # so we start with the configs for mpich2:
-                env.set("ESMF_COMM", "mpich2")
-                # The mpich 3 series split apart the Fortran and C bindings,
-                # so we link the Fortran libraries when building C programs:
-                env.set("ESMF_CXXLINKLIBS", "-lmpifort")
+                if self.version < Version("8.3.0"):
```

Comment:
When checking the version of the current package, `self.spec.satisfies()` should be used (see https://spack-tutorial.readthedocs.io/en/latest/tutorial_packaging.html#querying-spec-versions).

Reply:
Thanks for the tip. Two places were updated to follow the recommendation.